Automattic/woocommerce-payments
# Get WooPay adapted extensions from server

#### Changes proposed in this Pull Request








This PR updates the WooPay incompatible extensions, adapted extensions and available countries list from server with a single request.

This PR needs to be tested alongside 3568-gh-Automattic/woocommerce-payments-server.

#### Testing instructions





* Follow 3568-gh-Automattic/woocommerce-payments-server testing instructions.
* Follow [#6608](https://github.com/Automattic/woocommerce-payments/pull/6608) testing instructions changing `validate_incompatible_extensions` to `validate_woopay_compatibility` to ensure everything about incompatible extensions is working.
* Install the Points & Rewards plugin.
* Modify the lines [here](https://github.com/Automattic/woocommerce-payments-server/pull/3568/files#diff-5bb125880899ef7de4311905c5552402c73c1d7640a6c6e9d8bca2447ec86c0fR71) to look like this: `'adapted_extensions' => ['woocommerce-points-and-rewards'], 'available_countries' => ['US', 'BR'],` 
* Run `validate_woopay_compatibility` on the cron plugin.
* Access [phpMyAdmin](http://localhost:8083/index.php?route=/table/sql&db=wordpress&table=wp_options) and run this query:

```
SELECT * FROM `wp_options` WHERE `option_name` = 'woocommerce_woocommerce_payments_woopay_available_countries'
```

* The `meta_value` should match the value you filled out.
* Run this query:

```
SELECT * FROM `wp_options` WHERE `option_name` = 'woopay_enabled_adapted_extensions'
```

* The `meta_value` should contain `woocommerce-points-and-rewards`.
* Uninstall or deactivate Points & Rewards plugin.
* Run `validate_woopay_compatibility` on the cron plugin.
* Run this query on phpMyAdmin:

```
SELECT * FROM `wp_options` WHERE `option_name` = 'woopay_enabled_adapted_extensions'
```

* The `meta_value` should not contains `woocommerce-points-and-rewards`.
* Deactivating or activating Points & Rewards plugin without running the cron should also update the `woopay_enabled_adapted_extensions` value.

-------------------


**Post merge**
